optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -112,6 +111,7 @@
     @XStreamConverter(KeyAsElementMapConverter.class)
     protected Map<String, String> incrementalScoreCalculatorCustomProperties = null;
 
+    @Deprecated(/* forRemoval = true */)
     protected String ksessionName = null;
```

Comment:
We should be careful here.
If for some reason we have to make 8 compatible with 7 tooling, we can not remove this.

Reply:
Yea, I don't think we can deprecate this just yet. There's no good alternative today (maybe quarkus extension but still we can't deprecate kie-server in 7.x).
Please undeprecate, but add a "// TODO remove for 8.0"